src/requisition-convert-to-order/convert-to-order.controller.js

```diff
@@ -0,0 +1,410 @@
+         * Responsible for marking/unmarking all requisitions as selected.
+         *
+         * @param {Boolean} selectAll Determines if all requisitions should be selected or not
+         */
+        function toggleSelectAll(selectAll) {
+            angular.forEach(vm.requisitions, function(requisition) {
+                requisition.$selected = selectAll;
+                vm.onRequisitionSelect(requisition);
+            });
+        }
+
+        /**
+         * @ngdoc method
+         * @methodOf requisition-convert-to-order.controller:ConvertToOrderController
+         * @name setSelectAll
+         *
+         * @description
+         * Responsible for making the checkbox "select all" checked when all requisitions are
+         *     selected by user.
+         */
+        function setSelectAll() {
+            var value = true;
+            angular.forEach(vm.requisitions, function(requisition) {
+                value = value && requisition.$selected;
+            });
+            vm.selectAll = value;
+        }
+
+        /**
+         * @ngdoc method
+         * @methodOf requisition-convert-to-order.controller:ConvertToOrderController
+         * @name loadPreviouslySelectedRequisitions
+         *
+         * @description
+         * Selects checkboxes on current page if checked before
+         */
+        function loadPreviouslySelectedRequisitions() {
+            var storageRequisitions = $window.sessionStorage.getItem(vm.selectedRequisitionsStorageKey);
+            storageRequisitions = storageRequisitions ? JSON.parse(storageRequisitions) : {};
+
+            for (var i = 0; i < vm.requisitions.length; i++) {
+                var r = vm.requisitions[i].requisition;
+
+                if (storageRequisitions[r.id] !== undefined) {
+                    vm.requisitions[i].$selected = true;
+                }
+            }
+
+            setSelectAll();
+        }
+
+        /**
+         * @ngdoc method
+         * @methodOf requisition-convert-to-order.controller:ConvertToOrderController
+         * @name onRequisitionSelect
+         *
+         * @description
+         * Syncs requisition selection with storage
+         */
+        function onRequisitionSelect(requisition) {
+            var storageRequisitions = $window.sessionStorage.getItem(vm.selectedRequisitionsStorageKey);
+            
+            storageRequisitions = storageRequisitions ? JSON.parse(storageRequisitions) : {};
+
+            var requisitionId = requisition.requisition.id;
+
+            if (requisition.$selected) {
+                storageRequisitions[requisitionId] = requisition;
+            } else {
+                delete storageRequisitions[requisitionId];
+            }
+
+            $window.sessionStorage.setItem(
+                vm.selectedRequisitionsStorageKey, JSON.stringify(storageRequisitions)
+            );
+
+            setSelectAll();
+        }
+
+        // MALAWISUP-4161: Start: define new function
```

Comment:
This comment does not say much about code it's attached to.

Reply:
Done